### PR TITLE
Ports holomap blueprints from tg/vg.

### PR DESCRIPTION
**Pull request by tigercat2000**
https://github.com/tgstation/-tg-station/pull/16729
Changes:
 - Clicking "view structural data" in the station blueprints menu will
   show you ghost images of where things like disposal pipes, wires,
   atmospherics pipes, and other power machinery are originally supposed
   to go. Note, this will show you the ghost images over turfs even if the
   machinery is destroyed.

:cl:
rscadd: The station blueprints can now show you where wires/pipes/disposal pipes are supposed to go.
/:cl:

### CODE REVIEW

**Comment by Fox-McCloud** [comment]
Could you please port the borg blueprints and give it to the engineering borg (and prossibly drones)? They did this on TG to make their job of repairing easier.

**Comment by LittleBigKid2000** [comment]
Hey, I had this idea, you stole my idea that never left my head!

**Comment by tigercat2000** [comment]
@Fox-McCloud that's not appropriate for this PR and warrants further discussion about the implications of allowing the power to change area names to silicons

**Comment by pinatacolada** [comment]
i think i love you

**Comment by Dinarzad** [comment]
@tigercat2000 I don't THINK the Borg Blueprints can rename areas like the CE blueprints can. I think they literally are just the overlay aspects, so that they can repair things according to that layout. (Not really sure, I can't seem to find much on it in the wiki's an shit an I don't play TG)

**Comment by PJB3005** [comment]
@tigercat2000 or you port my PR on /vg/ in which this is a seperate item.

File: code/game/objects/objs.dm [hunk 1]
```diff
@@ -23,6 +23,19 @@
 	// Reagent ID => friendly name
 	var/list/reagents_to_log=list()
 
+	var/on_blueprints = FALSE //Are we visible on the station blueprints at roundstart?
```
Comment (PJB3005) [inline code/game/objects/objs.dm]: Sorry but TG's version of these vars is retarded, the one on /vg/ (is it on the holomap, and should it be automatic) is way better.

**Comment by monster860** [comment]
Bonus points if you make it show turfs too.

File: code/game/objects/objs.dm (hunk 1 above)
Comment (tigercat2000) [inline code/game/objects/objs.dm]: these vars are fine

**Comment by tigercat2000** [comment]
@Dinarzad they are fully functioning blueprints on tg

**Comment by ProperPants** [comment]
Can normal engineers see this too, or only someone holding a blueprint? Because it is phenomenally useful, but if only the CE can see the ghost pipes, they'd kinda have to do a lot by themselves.

Maybe in a future PR, to expand on this, an item like what @Dinarzad is saying could be made and given to normal engineers and borgs - a blueprint that is overlay only, not room changing.

Otherwise :+1: :heavy_multiplication_x: :100:

**Comment by tigercat2000** [comment]
@ProperPants it's tied directly to the blueprints 
i figure the CE can bother to come out of his office if anything explodes

**Comment by taukausanake** [comment]
It's not showing a scrubber, would that be an issue? And would the borgs get the blueprint as an upgrade module?

**Comment by tigercat2000** [comment]
@taukausanake the borgs aren't getting anything in this pr

**Comment by ProperPants** [comment]
@taukausanake You could probably guess where scrubbers or vents are by looking at where a pipe terminates, but it would be more convenient.

**Comment by Necaladun** [comment]
Very cool, I like it!

**Comment by fludd12** [comment]
Do you plan on / Do we already have the thing where blueprints can be photocopied to make them one-use prints that still have this feature? And, will this feature also apply to the slime blueprints?

**Comment by tigercat2000** [comment]
@fludd12 no and no

**Pull request merged**